src/hotspot/share/adlc/formsopt.cpp

```diff
@@ -172,8 +172,10 @@ int RegisterForm::RegMask_Size() {
   // that need more parameters will NOT be compiled.  On Intel, the limit
   // is something like 90+ parameters.
   // Add a few (3 words == 96 bits) for incoming & outgoing arguments to calls.
+  // Add one more word to avoid problematic rounding. Specifically, APX added
+  // 16 more registers but did not result in a mask size increase.
   // Round up to the next doubleword size.
```

Comment:
I agree with Dean that the comment should state more clearly what is problematic with the current rounding. From your analysis in the JBS issue, I take it that the problem is that
1) the number of bits left for stack locations depends on the value of `words_for_regs`,
2) before the addition of APX registers we were relying on the slack created by the up-rounding to accommodate enough stack locations, and
3) after adding APX registers this slack has been reduced significantly.

Is this the case? If so, I suggest to update the comment to something like:

```suggestion
  // Round up to the next doubleword size.
  // Add one more word to accommodate a reasonable number of stack locations
  // in the register mask regardless of how much slack is created by rounding up.
```

Reply:
I'm fine with this change, although I'd then argue that we could simplify it and just say we add 4 words (instead of 3) for incoming/outgoing arguments. @vnkozlov You suggested the current wording that specifically mentions APX, what do you think?

Reply:
Thanks @robcasloz and @vnkozlov, now updated.